imager_images/test_models2.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import unicode_literals
+from django.test import TestCase, override_settings
+import factory
+# from faker import Faker
+from faker import Factory as FakerFactory
+from factory import post_generation, lazy_attribute
+from django.contrib.auth.models import User
+from imager_images.models import Photo, Album
+import datetime
+import os
+import tempfile
+
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+TEST_PHOTO_PATH = os.path.join(HERE, "didrctoryt", "filename")
+TEST_MEDIA = tempfile.mkdtemp()
+
+fake = FakerFactory.create()
+
+
+class UserFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = User
+
+    username = factory.Sequence(lambda n: "user{}".format(n))
+    email = factory.Sequence(
+        lambda n: "user{}@example.com".format(n)
+    )
+
+
+class AlbumFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = Album
+
+    photographer = factory.SubFactory(UserFactory)
+
+    title = factory.lazy_attribute(lambda o: fake.sentence(nb_words=2))
+    description = factory.lazy_attribute(lambda o: fake.sentence(nb_words=5))
+
+
+class PhotoFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = Photo
+
+    photographer = factory.SubFactory(UserFactory)
+
+    file = factory.django.ImageField(color='red')
+    height_field = 100
+    width_field = 100
+
+
+class ModelTestCases(TestCase):
+
+    def setUp(self):
+        self.user = UserFactory.create()
+        self.client.force_login(user=self.user)
+        self.album = AlbumFactory(photographer=self.user)
+        self.today = datetime.date.today()
+
+    def test_album(self):
+        self.album = AlbumFactory(photographer=self.user)
+        self.client.force_login(user=self.user)
+        self.assertTrue(self.album.published == 'private')
+        self.assertTrue(type(self.album.title) == str)
+        self.assertEqual(self.album.date_created, self.today)
+        self.assertEqual(self.album.date_modified, self.today)
+        self.assertEqual(self.album.date_pub, None)
+        self.assertEqual(self.album.cover_photo, '')
+        self.assertTrue(self.album.description is not None)
```

Comment:
These are just testing all of the attributes of the album, but nothing else. What happens when an album is actually created? What happens when a photo is added to the album? What about when a cover photo is added? What do I see when I see a list of albums? Do I see them all or just the public ones? What happens when an album is saved without the proper input values? Is there anything that validates that? Think about how the app is going to be using this model, and write tests for that, not just the existence of the model.